ui/app/components/app/token-cell/token-cell.js

```diff
@@ -0,0 +1,92 @@
+import classnames from 'classnames'
+import PropTypes from 'prop-types'
+import React from 'react'
+import { conversionUtil, multiplyCurrencies } from '../../../helpers/utils/conversion-util'
+import Tooltip from '../../ui/tooltip-v2'
+import AssetListItem from '../asset-list-item'
+import { useSelector } from 'react-redux'
+import { getTokenExchangeRates, getConversionRate, getCurrentCurrency, getSelectedAddress } from '../../../selectors'
+import { useI18nContext } from '../../../hooks/useI18nContext'
+import InfoIcon from '../../ui/icon/info-icon.component'
+import { formatCurrency } from '../../../helpers/utils/confirm-tx.util'
+
+export default function TokenCell ({ address, outdatedBalance, symbol, string, image, onClick }) {
+  const contractExchangeRates = useSelector(getTokenExchangeRates)
+  const conversionRate = useSelector(getConversionRate)
+  const currentCurrency = useSelector(getCurrentCurrency)
+  const userAddress = useSelector(getSelectedAddress)
+  const t = useI18nContext()
+
+  let currentTokenToFiatRate
+  let currentTokenInFiat
+  let formattedFiat = ''
+
+
+  if (contractExchangeRates[address]) {
+    currentTokenToFiatRate = multiplyCurrencies(
+      contractExchangeRates[address],
+      conversionRate
+    )
+    currentTokenInFiat = conversionUtil(string, {
+      fromNumericBase: 'dec',
+      fromCurrency: symbol,
+      toCurrency: currentCurrency.toUpperCase(),
+      numberOfDecimals: 2,
+      conversionRate: currentTokenToFiatRate,
+    })
+    formattedFiat = `${formatCurrency(currentTokenInFiat, currentCurrency)} ${currentCurrency.toUpperCase()}`
```

Comment:
Previously this would omit fiat altogether if the value was zero, e.g. `$0`.

I think that's still something we want to do? 🤔

Reply:
Speaking only from my personal preference: I like having it say $0.00 when token quantity is 0. It creates symmetry with other non zero balance tokens.

Reply:
Even when the conversion rate is missing? A side-effect of the old logic was that the fiat was shown in the case where the information required to convert it was missing.

Showing $0.00 for a token worth a non-zero amount does seem misleading. Perhaps you could update this to hide the fiat balance if the conversion rate is missing, instead of when it's zero.

Reply:
if the conversion rate is missing, currentTokenInFiat never gets set. My assumption was showFiat would be false when it tries to evaluate `Boolean(currentTokenInFiat)` in this case. Is that not correct?

Reply:
oh, are you saying `conversionRate` is missing not `contractExchangeRates[address]`?

Reply:
Yeah, `currentTokenToFiatRate` in particular. I thought that the conversion rate defaulted to zero if it wasn't given? 🤔 Maybe that's not the case.

Reply:
Ah, apparently it's always fetched but it starts out as zero: https://github.com/MetaMask/controllers/blob/develop/src/assets/CurrencyRateController.ts#L79

Reply:
Ah, got it. What if i changed:

```
if (contractExchangeRates[address]) {
```

to

```
if (conversionRate > 0 && contractExchangeRates[address]) {
```

Reply:
That seems reasonable!